<a href="https://colab.research.google.com/github/dhairyaostwal/CSE4020-Codes/blob/main/Cancer_Data_Reform.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import pandas as pd

# Combining Columns

1. WBC_area,	WBC_convex_area
2. nuc_area, convex_area_nuc

In [ ]:
target = data['Diagnosis']
target

0    M
1    B
2    M
3    M
4    M
5    B
6    B
7    M
8    M
9    B
Name: Diagnosis, dtype: object

In [ ]:
WBC_area, WBC_convex_area = data['WBC_area'], data['WBC_convex_area']
nuc_area, convex_area_nuc = data['nuc_area'], data['convex_area_nuc']

In [ ]:
WBC_area

0    10073
1      450
2    10349
3    10827
4     3421
5     9994
6     4497
7     2614
8    15374
9    10429
Name: WBC_area, dtype: int64

In [ ]:
WBC_convex_area

0    10170
1      450
2    10506
3    10862
4     3470
5    10131
6     4680
7     2680
8    15509
9    10465
Name: WBC_convex_area, dtype: int64

In [ ]:
WBC_Area_New = round((WBC_area + WBC_convex_area)/2)
WBC_Area_New

0    10122.0
1      450.0
2    10428.0
3    10844.0
4     3446.0
5    10062.0
6     4588.0
7     2647.0
8    15442.0
9    10447.0
dtype: float64

In [ ]:
print(WBC_Area_New)

print(target)

0    10122.0
1      450.0
2    10428.0
3    10844.0
4     3446.0
5    10062.0
6     4588.0
7     2647.0
8    15442.0
9    10447.0
dtype: float64
0    M
1    B
2    M
3    M
4    M
5    B
6    B
7    M
8    M
9    B
Name: Diagnosis, dtype: object


In [ ]:
# as we can see mean doesn't work therefore trying out with median, mode

In [ ]:
nuc_area_new = round((nuc_area + convex_area_nuc)/2)
nuc_area_new

0     7800.0
1      258.0
2     8162.0
3     8460.0
4     2604.0
5     7527.0
6     3192.0
7     2184.0
8    11201.0
9     7670.0
dtype: float64

In [ ]:
target

0    M
1    B
2    M
3    M
4    M
5    B
6    B
7    M
8    M
9    B
Name: Diagnosis, dtype: object

In [ ]:
data2 = {'nuc_area': nuc_area, 
        'convex_area_nuc': convex_area_nuc,
        'nuc_area_merged': nuc_area_new,
         'target': target}

df = pd.DataFrame(data2)
df         

nuc_area  convex_area_nuc  nuc_area_merged target
0      7670             7930           7800.0      M
1       204              312            258.0      B
2      8092             8232           8162.0      M
3      8326             8593           8460.0      M
4      2582             2627           2604.0      M
5      6968             8086           7527.0      B
6      2682             3701           3192.0      B
7      2138             2229           2184.0      M
8     10647            11755          11201.0      M
9      7555             7786           7670.0      B

In [ ]:
df.sort_values(by=['nuc_area_merged'])

nuc_area  convex_area_nuc  nuc_area_merged target
1       204              312            258.0      B
7      2138             2229           2184.0      M
4      2582             2627           2604.0      M
6      2682             3701           3192.0      B
5      6968             8086           7527.0      B
9      7555             7786           7670.0      B
0      7670             7930           7800.0      M
2      8092             8232           8162.0      M
3      8326             8593           8460.0      M
8     10647            11755          11201.0      M

## Re uploading data(Session Expired)

In [13]:
data = pd.read_csv('Data.csv')

In [14]:
data

Id  WBC_area  WBC_convex_area  ...  axismeanratio  Diagnosis  Unnamed: 24
0   1     10073            10170  ...       1.718868          M          NaN
1   2       450              450  ...       1.586649          B          NaN
2   3     10349            10506  ...       1.728577          M          NaN
3   4     10827            10862  ...       1.721235          M          NaN
4   5      3421             3470  ...       1.769519          M          NaN
5   6      9994            10131  ...       1.710901          B          NaN
6   7      4497             4680  ...       1.774326          B          NaN
7   8      2614             2680  ...       1.819403          M          NaN
8   9     15374            15509  ...       1.670522          M          NaN
9  10     10429            10465  ...       1.678827          B          NaN

[10 rows x 25 columns]

## Trying with another features

In [16]:
avg_cyt_re, avg_cyt_gr, avg_cyt_bl = data['avg_cyt_re'], data['avg_cyt_gr'], data['avg_cyt_bl']

target = data['Diagnosis']



In [18]:
data2 = {'avg_cyt_re': avg_cyt_re,
         'avg_cyt_gr': avg_cyt_gr,
         'avg_cyt_bl': avg_cyt_bl,
         'target': target}

df = pd.DataFrame(data2)
df

avg_cyt_re  avg_cyt_gr  avg_cyt_bl target
0  130.713275  123.667915  145.271328      M
1  115.975610   67.243902  140.126016      B
2  135.955693  129.935312  144.863093      M
3  131.139544  122.655338  143.148741      M
4  150.808105  142.667461  176.097735      M
5  130.945142   91.728024  147.307336      B
6   90.649587   58.960331   96.546006      B
7  132.686975  126.090336  151.857143      M
8  143.154220  131.805585  163.624921      M
9  126.637787  112.096381  143.146486      B

### Sorting by **avg_cyt_re**

In [19]:
df.sort_values(by=['avg_cyt_re'])

avg_cyt_re  avg_cyt_gr  avg_cyt_bl target
6   90.649587   58.960331   96.546006      B
1  115.975610   67.243902  140.126016      B
9  126.637787  112.096381  143.146486      B
0  130.713275  123.667915  145.271328      M
5  130.945142   91.728024  147.307336      B
3  131.139544  122.655338  143.148741      M
7  132.686975  126.090336  151.857143      M
2  135.955693  129.935312  144.863093      M
8  143.154220  131.805585  163.624921      M
4  150.808105  142.667461  176.097735      M

### Sorting by **target**

In [20]:
df.sort_values(by=['target'])

avg_cyt_re  avg_cyt_gr  avg_cyt_bl target
1  115.975610   67.243902  140.126016      B
5  130.945142   91.728024  147.307336      B
6   90.649587   58.960331   96.546006      B
9  126.637787  112.096381  143.146486      B
0  130.713275  123.667915  145.271328      M
2  135.955693  129.935312  144.863093      M
3  131.139544  122.655338  143.148741      M
4  150.808105  142.667461  176.097735      M
7  132.686975  126.090336  151.857143      M
8  143.154220  131.805585  163.624921      M

As we can see out of **`avg_cyt_bl`, `avg_cyt_gr`, `avg_cyt_re`** only **`avg_cyt_gr`** gives proper range of following:

**B = [67.24 : 112.09]**

**M = [122.655338 : 142.667461:]**